example/e2e/helpers.ts

```diff
@@ -0,0 +1,218 @@
+    try {
+      for (let i = 0; i < 3; i++) {
+        await element(by.text(Selectors.THREEDS2_COMPLETE_BUTTON)).longPress();
+        await new Promise((resolve) => setTimeout(resolve, 1000));
+      }
+    } catch (exception) {}
+  } else {
+    await waitFor(element(by.text(Selectors.THREEDS2_SCREEN_HEADER)))
+      .toBeVisible()
+      .withTimeout(15000);
+    await element(by.text(Selectors.THREEDS2_COMPLETE_BUTTON)).longPress();
+    await new Promise((resolve) => setTimeout(resolve, 5000));
+    await element(by.text(Selectors.THREEDS2_COMPLETE_BUTTON)).longPress();
+  }
+}
+
+export async function isCardAddedToPaymentMethods(): Promise<boolean> {
+  try {
+    await expect(element(by.label(Selectors.EXISTING_CARD))).toBeVisible();
+    return true;
+  } catch (exception) {
+    return false;
+  }
+}
+
+export async function addCardPaymentMethodAndPay() {
+  if (await isCardAddedToPaymentMethods()) {
+    await element(by.label(Selectors.EXISTING_CARD)).swipe('left');
+    await element(by.text('Delete')).tap();
+    await new Promise((resolve) => setTimeout(resolve, 3000));
+  }
+  if (device.getPlatform() === 'android') {
+    await element(by.text(Selectors.ADD_CARD_BUTTON)).tap();
+  } else {
+    await element(by.text(Selectors.ADD_CARD_BUTTON)).atIndex(1).tap();
+  }
+  await fillPaymentDetailsSheet({
+    number: TestData.CARD_NUMBER,
+    name: TestData.CARDHOLDER_NAME,
+    expiry: TestData.EXPIRY_DATE,
+    code: TestData.SECURITY_CODE,
+  });
+  if (device.getPlatform() === 'android') {
+    await new Promise((resolve) => setTimeout(resolve, 2000));
+    await device.disableSynchronization();
+    await element(by.id(Selectors.ANDROID_METHODS_PAY_NOW)).tap();
+  } else {
+    await element(by.text('PAY NOW')).tap();
+  }
+  await fillSecurityCodeSheet();
+  await tapPayNowButton();
+  await complete3DS2();
+}
+
+export async function setNoPreferenceCRI() {
+  if (device.getPlatform() === 'android') {
+    await device.enableSynchronization();
+  }
+  await element(by.id(Selectors.SETTINGS_BUTTON)).tap();
+  await element(by.text(Selectors.CHALLENGE_REQUEST_SETTINGS)).tap();
+  await element(by.text(Selectors.NO_PREFERENCE)).tap();
+  if (device.getPlatform() === 'android') {
+    try {
+      await expect(element(by.id(Selectors.AUTH_TOGGLE))).toHaveToggleValue(
+        true
+      );
+    } catch (exception) {
+      await enterAuthDetails();
+    }
+  } else {
+    let tokenToggle = await element(
+      by.id(Selectors.AUTH_TOGGLE)
+    ).getAttributes();
+    if ('value' in tokenToggle) {
+      if (tokenToggle.value === '0') {
+        await enterAuthDetails();
+      }
+    }
+  }
+  if (device.getPlatform() === 'android') {
+    await device.disableSynchronization();
+  }
+}
+
+export async function clickSettingsButton() {
+  if (device.getPlatform() === 'ios') {
+    await element(by.id(Selectors.SETTINGS_BUTTON)).tap();
+  } else if (device.getPlatform() === 'android') {
```

Comment:
There are lots of places where these checks are happening. Can we do something like:

```javascript
function isIOS(): boolean {
  return device.getPlatform === 'ios';
}

function isAndroid(): boolean {
  return device.getPlatform() === 'android';
}
```

And then: 
```javascript
if (isIOS()) {
} else if (isAndroid()) {
}
```

Reply:
Agreed - refactored